### PR TITLE
chore(release): update changelog and bump version 0.20.0

### PR DESCRIPTION
**Pull request by jawid-h**
## Issue being fixed or feature implemented

Release 0.20.0

## What was done?

- Update changelog
- Bump version to 0.20.0

## How Has This Been Tested?

None

## Breaking Changes

None

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,17 @@
+# [0.20.0](https://github.com/dashevo/js-dpp/compare/v0.19.2...v0.20.0) (2021-07-08)
+
+
+### Features
+
+* add metadata to document, data contract and identity ([#318](https://github.com/dashevo/js-dpp/issues/318))
+* update @dashevo/re2-wasm to 1.0.4 ([#317](https://github.com/dashevo/js-dpp/issues/317))
```
Comment (shumkov) [inline CHANGELOG.md:7]: Combine with re2

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,17 @@
...
+* initialization will run only once ([#315](https://github.com/dashevo/js-dpp/issues/315))
+* update JSON Schema to draft 2020-12 ([#312](https://github.com/dashevo/js-dpp/issues/312))
+* replace RegExp with RE2 ([#311](https://github.com/dashevo/js-dpp/issues/311))
+* integrate AJV v7 ([#310](https://github.com/dashevo/js-dpp/issues/310))
```
Comment (shumkov) [inline CHANGELOG.md:11]: Strcit Data contract JSON schema validation

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,17 @@
+# [0.20.0](https://github.com/dashevo/js-dpp/compare/v0.19.2...v0.20.0) (2021-07-08)
+
+
+### Features
+
+* add metadata to document, data contract and identity ([#318](https://github.com/dashevo/js-dpp/issues/318))
+* update @dashevo/re2-wasm to 1.0.4 ([#317](https://github.com/dashevo/js-dpp/issues/317))
+* initialization will run only once ([#315](https://github.com/dashevo/js-dpp/issues/315))
+* update JSON Schema to draft 2020-12 ([#312](https://github.com/dashevo/js-dpp/issues/312))
+* replace RegExp with RE2 ([#311](https://github.com/dashevo/js-dpp/issues/311))
```
Comment (shumkov) [inline CHANGELOG.md:10]: Prevent possible ReDoS attacks for Data Contracts

**Review by shumkov — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,17 @@
+# [0.20.0](https://github.com/dashevo/js-dpp/compare/v0.19.2...v0.20.0) (2021-07-08)
+
+
+### Features
+
+* add metadata to document, data contract and identity ([#318](https://github.com/dashevo/js-dpp/issues/318))
+* update @dashevo/re2-wasm to 1.0.4 ([#317](https://github.com/dashevo/js-dpp/issues/317))
+* initialization will run only once ([#315](https://github.com/dashevo/js-dpp/issues/315))
```
Comment (shumkov) [inline CHANGELOG.md:8]: Combine with ReDoS

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,17 @@
...
+* update JSON Schema to draft 2020-12 ([#312](https://github.com/dashevo/js-dpp/issues/312))
+* replace RegExp with RE2 ([#311](https://github.com/dashevo/js-dpp/issues/311))
+* integrate AJV v7 ([#310](https://github.com/dashevo/js-dpp/issues/310))
+
```
Comment (shumkov) [inline CHANGELOG.md:12]: Describe breaking changes

**Review by shumkov — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,27 @@
...
+### BREAKING CHANGES
+
+* `dependencies` is not supported. Use `dependentRequired` and `dependentSchema` instead
+* `additionalitems` is not supported. Use `items: false` and `prefixItems` instead
+* `patternProperties` is prohibited for Data Contract
+* error messages and properties are changed according to the new JSON Schema spec
+* `pattern` keyword accept only RE2 compatible regular expressions
+* Document type and properties minimum length is 3 chars
+* `definitions` is now defined using `$defs` keyword
+* JSON Schema strict validation is enabled . Previous contract schemas invalid in case they are not respecting strict mode constraints (reference this link for more information)
```
Comment (thephez) [inline CHANGELOG.md:20]: "reference this link for more information" doesn't actually include a link to something

Comment (jawid-h) [reply CHANGELOG.md]: fixed, thank you

**Review by shumkov — APPROVED** [review]
👍

**Pull request merged**